### PR TITLE
Fixes #25523 - Fallback to english if js locale is missing

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: webpack/assets/javascripts/react_app/common/I18n.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const locales = window.locales || emptyLocales;
 let locale = document.getElementsByTagName('html')[0].lang;
 
 locale = locale.length === 0 ? 'en' : locale.replace(/-/g, '_');
-export const jed = new Jed(locales[locale]);
+export const jed = new Jed(locales[locale] || locales['en']);
```
Comment (ohadlevy) [inline webpack/assets/javascripts/react_app/common/I18n.js]: does it make sense to log this to somewhere?

Comment (tbrisker) [reply webpack/assets/javascripts/react_app/common/I18n.js]: updated to log failures to the console

File: webpack/assets/javascripts/react_app/common/I18n.js [hunk 2]
```diff
@@ -12,6 +12,13 @@ const locales = window.locales || emptyLocales;
 let locale = document.getElementsByTagName('html')[0].lang;
 
 locale = locale.length === 0 ? 'en' : locale.replace(/-/g, '_');
+
+if (locales[locale] === undefined) {
+  // eslint-disable-next-line no-console
+  console.log(`could not load translations for ${locale} locale, falling back to English`);
+  locale = 'en';
+}
```
Comment (ohadlevy) [inline webpack/assets/javascripts/react_app/common/I18n.js]: last nitpick if you may, could you extract 'en' as a const e.g.
```js
const DEFAULT_LOCALE='en';

locale = locale.length === 0 ? DEFAULT_LOCALE : locale.replace(/-/g, '_');
￼
if (locales[locale] === undefined) {
￼   // eslint-disable-next-line no-console
￼   console.log(`could not load translations for ${locale} locale, falling back to English`);
￼   locale = DEFAULT_LOCALE;
￼ }

Comment (tbrisker) [reply webpack/assets/javascripts/react_app/common/I18n.js]: changed

File: webpack/assets/javascripts/react_app/common/I18n.js [hunk 3]
```diff
@@ -4,14 +4,21 @@ import { deprecateObjectProperty } from '../../foreman_tools';
...
+const DEFAULT_LOCALE = 'en';
+
+const emptyLocales = {};
+emptyLocales[DEFAULT_LOCALE] = { domain: 'app', locale_data: { app: { '': {} } } };
 
 const locales = window.locales || emptyLocales;
 let locale = document.getElementsByTagName('html')[0].lang;
+locale = locale.length === 0 ? DEFAULT_LOCALE : locale.replace(/-/g, '_');
+
+if (locales[locale] === undefined) {
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/common/I18n.js]: can you add a test for this?

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @tbrisker ! just a minor request

File: webpack/assets/javascripts/react_app/common/I18n.js (hunk 3 above)
Comment (tbrisker) [reply webpack/assets/javascripts/react_app/common/I18n.js]: I tried, but testing non-exported functions or constructor methods is non trivial. I did refactor the code a bit in hope of making it testable, but creating a good test for it is proving to be too much overhead for such change.

**Review by ohadlevy — APPROVED** [review]

**Comment by tbrisker** [comment]
cherry-picked to 1.20-stable as c82e315fd in the hopes that it doesn't break webpack for plugins :crossed_fingers:

**Pull request merged**